src/components/modals/data-table-modal.js

```diff
@@ -30,183 +32,179 @@ import {Clock} from 'components/common/icons/index';
-    padding-left: ${dgSettings.sidePadding};
-  }
-  .react-grid-Cell:last-child,
-  .react-grid-HeaderCell:last-child {
-    padding-right: ${dgSettings.sidePadding};
-  }
-  .react-grid-Cell__value {
-    color: ${props => props.theme.labelColorLT};
-  }
-  .react-grid-Canvas {
-    ${props => props.theme.modalScrollBar};
+  .cell {
+    padding: 14px 8px;
+    border-right: 0;
+    border-bottom: ${props => props.theme.panelBorderLT};
   }
-  
-  ${media.palm`
-    .react-grid-Container {
-      /* TODO: replace data-grid component with react-window */
-      /* We need to use important in this case to override data-grid styling */
-      width: 100vw !important;
-    }
-  `};
-  
+ 
 `;
-const StyledModal = styled.div`
-  ${media.palm`
-    margin: 0 -36px;
-  `}
-`;
-const BooleanFormatter = ({value}) => <span>{String(value)}</span>;
 
-export class DataTableModal extends Component {
-
-  constructor(props) {
-    super(props);
-    this._root = React.createRef();
-  }
+const tagContainerStyle = {
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'space-between',
+  alignItems: 'center'
+};
 
-  _onMouseWheel = e => {
-    // Prevent futile scroll, which would trigger the Back/Next page event
-    // https://github.com/micho/jQuery.preventMacBackScroll
-    // This prevents scroll when reaching the topmost or leftmost
-    // positions of a container.
+const HeaderRenderer = React.memo(({columnIndex, data, rowIndex, style}) => {
+  const item = data[columnIndex];
 
-    // react-data-grid canvas element can be scrolled
-    const canvas = this._root.querySelector('.react-grid-Canvas');
+  return (
+    <div className="header" style={{...style, ...tagContainerStyle}}>
+      <div style={{display: 'flex', alignItems: 'center'}}>
+        <div>
+          {item.type === 'timestamp' ? <Clock height="16px" /> : null}
+        </div>
+        {item.name}
+      </div>
+      <div>
+        <FieldToken type={item.type} />
+      </div>
+    </div>
+  );
+});
+
+const CellRenderer = React.memo(({columnIndex, data, rowIndex, style}) => {
+  const item = data[rowIndex][columnIndex];
+
+  return (
+    <div
+      className="cell"
+      style={style}
+      key={`${rowIndex}-${columnIndex}`}>
+      {item}
+    </div>
+  );
+});
 
-    // If canvas can not be scrolled left anymore when we try to scroll left
-    const prevent_left = e.deltaX < 0 && canvas.scrollLeft <= 0;
-    // If canvas can not be scrolled up when we try to scroll up
-    const prevent_up = e.deltaY < 0 && canvas.scrollTop <= 0;
+export class DataTableModal extends Component {
 
-    if (prevent_left || prevent_up) {
-      e.preventDefault();
-    }
-  };
+  headerGrid = React.createRef();
+
+  datasetsSelector = props => props.datasets;
+  dataIdSelector = props => props.dataId;
+  activeDatasetSelector = createSelector(
+    this.datasetsSelector,
+    this.dataIdSelector,
+    (datasets, dataId) => datasets[dataId]
+  );
+  columnsSelector = createSelector(
```

Comment:
You don't need to call `createSelector` on `activeDatasetSelector` it is not derived data, it simply returns `datasets[id]`  This can just be
```js
activeDatasetSelector = props => props.datasets[props.dataId],
columnsSelector = createSelector(
    this.activeDatasetSelector,
    dataset => dataset.fields.filter(({name}) => name !== '_geojson')
  );
```